File: ueueo-framework/ueueo-authorization-abstractions/src/main/java/com/ueueo/authorization/PermissionsRequirementHandler.cs
using System.Linq;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Volo.Abp.Authorization.Permissions;

namespace Volo.Abp.Authorization;

public class PermissionsRequirementHandler : AuthorizationHandler<PermissionsRequirement>
{
    private readonly IPermissionChecker _permissionChecker;

    public PermissionsRequirementHandler(IPermissionChecker permissionChecker)
    {
        _permissionChecker = permissionChecker;
    }

    protected override void HandleRequirementAsync(
        AuthorizationHandlerContext context,
        PermissionsRequirement requirement)
    {
        var multiplePermissionGrantResult = _permissionChecker.IsGrantedAsync(context.User, requirement.PermissionNames);

        if (requirement.RequiresAll ?
            multiplePermissionGrantResult.AllGranted :
            multiplePermissionGrantResult.Result.Any(x => x.Value == PermissionGrantResult.Granted))
        {
            context.Succeed(requirement);
        }
    }
}

File: ueueo-framework/ueueo-authorization-abstractions/src/main/java/com/ueueo/authorization/PermissionsRequirementHandler.java
package com.ueueo.authorization;

import com.ueueo.authorization.microsoft.AuthorizationHandler;
import com.ueueo.authorization.microsoft.AuthorizationHandlerContext;
import com.ueueo.authorization.permissions.IPermissionChecker;
import com.ueueo.authorization.permissions.MultiplePermissionGrantResult;
import com.ueueo.authorization.permissions.PermissionGrantResult;

public class PermissionsRequirementHandler extends AuthorizationHandler<PermissionsRequirement> {

    private final IPermissionChecker permissionChecker;

    public PermissionsRequirementHandler(IPermissionChecker permissionChecker) {
        this.permissionChecker = permissionChecker;
    }

    @Override
    protected void handleRequirement(AuthorizationHandlerContext context, PermissionsRequirement requirement) {
        MultiplePermissionGrantResult multiplePermissionGrantResult = permissionChecker.isGranted(context.getUser(), requirement.getPermissionNames());

        if (requirement.isRequiresAll() ?
                multiplePermissionGrantResult.allGranted() :
                multiplePermissionGrantResult.getResult().values().stream().anyMatch(PermissionGrantResult.Granted::equals)) {
            context.succeed(requirement);
        }
    }
}
